actionmailbox/lib/action_mailbox/engine.rb

```diff
@@ -20,13 +20,16 @@ class Engine < Rails::Engine
     config.action_mailbox.queues = ActiveSupport::InheritableOptions.new \
       incineration: :action_mailbox_incineration, routing: :action_mailbox_routing
 
+    config.action_mailbox.storage_service = nil
```

Comment:
Instead of nil shouldn't this be `config.action_mailbox.storage_service = config.active_storage.service`? That would restore the current behavior and allow opting in to re-configuring AMB?

Reply:
I can do that, too.  My rationale for nil was that nil was what InboundEmail was getting there in the first place before this patch.

Reply:
Just to clarify: passing nil was a deliberate choice to do exactly what InboundEmail was doing; that's how it was getting a default service. When the service was set to nil, it was getting the default service. 

I still suspect nil might be a better choice. If it's not nil, but (as you suggest) is copied from ActiveStorage configuration, then if AvtiveStorage default service is somehow overriden, then it won't follow the change. Right now, it will, as it just relies on ActiveStorage.service and doesn't care about ActiveStorage's configuration
My patch preserves this exact behavior of using whatever is the default service.

Just my 2 cents.

Reply:
Can you write a test to verify this? I think your rationale checks out but it would be good to be sure.

Reply:
> Can you write a test to verify this? I think your rationale checks out but it would be good to be sure.

How about this test below? If you uncomment a configuration like (find comment with `Uncomment` in it) the test will start failing, indicating that my rationale does indeed check out -- that setting this setting to `nil` allows us to adhere to the previous behaviour of using `ActiveStorage::Blob.service` as a default, not `ActiveStorage`'s configuration.

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "rails", path: "."
  gem "sqlite3"
  if RUBY_VERSION >= "3.1"
    # net-smtp, net-imap and net-pop were removed from default gems in Ruby 3.1, but is used by the `mail` gem.
    # So we need to add them as dependencies until `mail` is fixed: https://github.com/mikel/mail/pull/1439
    gem "net-smtp", require: false
  end
end

require "active_record/railtie"
require "active_storage/engine"
require "action_mailbox/engine"
require "tmpdir"

class TestApp < Rails::Application
  config.root = __dir__
  config.hosts << "example.org"
  config.eager_load = false
  config.session_store :cookie_store, key: "cookie_store_key"
  secrets.secret_key_base = "secret_key_base"

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger

  config.active_storage.service = :local
  config.active_storage.service_configurations = {
    local: {
      root: Dir.tmpdir,
      service: "Disk"
    },
    email: {
      root: "#{Dir.tmpdir}/email",
      service: "Disk"
    }
  }

  # Uncomment the line below to follow active_storage
  # config.action_mailbox.storage_service = config.active_storage.service

  config.action_mailbox.ingress = :relay
end

ENV["DATABASE_URL"] = "sqlite3::memory:"

Rails.application.initialize!

require ActiveStorage::Engine.root.join("db/migrate/20170806125915_create_active_storage_tables.rb").to_s
require ActionMailbox::Engine.root.join("db/migrate/20180917164000_create_action_mailbox_tables.rb").to_s

ActiveRecord::Schema.define do
  CreateActiveStorageTables.new.change
  CreateActionMailboxTables.new.change
end

class ApplicationMailbox < ActionMailbox::Base
end

require "minitest/autorun"

class RepliesMailboxTest < ActionMailbox::TestCase

  test "should follow current default storage service" do
    # ActiveStorage is configured to local
    assert_equal(:local, ActiveStorage::Blob.service.name)
    # Let's change it
    ActiveStorage::Blob.service = ActiveStorage::Blob.services.fetch(:email)
    assert_equal(:email, ActiveStorage::Blob.service.name)

    message = create_inbound_email_from_mail(from: "david@loudthinking.com", subject: "Hello!")
    raw = message.raw_email
    assert(ActiveStorage::Blob.services.fetch(:email).exist?(raw.key))
    assert_not(ActiveStorage::Blob.services.fetch(:local).exist?(raw.key))
  end

end
```

Reply:
Thanks, that looks good. I added a suggestion just below on how you could get a test for this into Rails, hopefully it works as I expect it to.

Reply:
I've followed your suggestion and simplified the test.

Reply:
@yrashk I can't see this test - are you planning to add it?

Reply:
@pixeltrix the test above was to verify my rationale on following default vs configuration.  The test I added in the commit does pretty much the same thing except that it doesn't test it with configuration setting as per @ghiculescu 's suggestion.

Thoughts?

Reply:
We probably need some integration tests around the configuration in https://github.com/rails/rails/blob/main/railties/test/application/configuration_test.rb

1. Test that the default configuration remains the default Active Storage service
2. Test that changing the default service after initialisation is picked up by Action Mailbox
3. Test that changing the Action Mailbox storage service works

Reply:
@pixeltrix I've amended the patch to add more tests as per your suggestion. Please let me know what you think.

Reply:
Ok I also see this is the behavior of ActiveStorage when `service_name` is nil: https://github.com/rails/rails/blob/bead3221c706e468e3625b71b7e99e9fd333a80c/activestorage/app/models/active_storage/blob.rb#L52